src/app/api/boosts/route.ts

```diff
@@ -0,0 +1,132 @@
+import { createClient } from '@supabase/supabase-js';
+import { NextResponse, type NextRequest } from 'next/server';
+import '@/utils/helper';
+import { hoursToMilliseconds } from 'date-fns';
+
+const supabase = createClient(
+    process.env.SUPABASE_URL as string,
+    process.env.SUPABASE_ANON_KEY as string
+);
+
+export type BoostType = {
+  id: bigint;
+  name: string;
+  contract_addresses: string[];
+  image_url: string | null;
+  game_id: bigint;
+  cta_url: string | null;
+  cta_text: string | null;
+  boost_type: string;
+  is_enabled: boolean;
+  points: bigint;
+  nft_amount: bigint | null;
+};
+
+export type Boost = {
+  id: bigint;
+  name: string;
+  contractAddresses: string[];
+  imageUrl: string | null;
+  gameId: bigint;
+  ctaUrl: string | null;
+  ctaText: string | null;
+  boostType: string;
+  isEnabled: boolean;
+  points: bigint;
+  nftAmount: bigint | null;
+};
+
+async function mapBoostState(
+  boost: BoostType
+): Promise<Boost> {
+  return {
+    id: boost.id as bigint,
+    name: boost.name,
+    contractAddresses: boost.contract_addresses,
+    imageUrl: boost.image_url,
+    gameId: boost.game_id as bigint,
+    ctaUrl: boost.cta_url,
+    ctaText: boost.cta_text,
+    boostType: boost.boost_type,
+    isEnabled: boost.is_enabled,
+    points: boost.points,
+    nftAmount: boost.nft_amount
+  };
+}
+
+export async function GET(request: NextRequest) {
+  const searchParams = request.nextUrl.searchParams;
+  const userAddress = searchParams.get('userAddress') as string;
+  const gameId = BigInt(searchParams.get('gameId') as string);
```

Comment:
Is this check working for you ? I had to seperate into 2 checks:

const gameIdString = searchParams.get("gameId") as string
const gameId = BigInt(gameIdString);

Reply:
you can rebase and use `toBigInt():bigint|null` and check if `gameId === null`